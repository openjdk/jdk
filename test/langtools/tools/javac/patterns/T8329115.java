/**
 * @test /nodynamiccopyright/
 * @bug 8329115
 * @summary Crash involving return from inner switch
 * @compile/fail/ref=T8329115.out -XDrawDiagnostics -XDdev T8329115.java
 */
public class T8329115 {
    record R1() {}
    record R2() {}

    int test() {
        return switch (new R1()) {
            case R1() -> {
                return switch (new R2()) { // crashes, instead it should just be the error: attempt to return out of a switch expression
                    case R2() -> 1;
                };
            }
        };
    }
}
