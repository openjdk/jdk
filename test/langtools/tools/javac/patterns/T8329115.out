T8329115.java:14:17: compiler.err.return.outside.switch.expression
T8329115.java:12:16: compiler.err.switch.expression.no.result.expressions
2 errors
